gridfs/gridfs.c

```diff
@@ -742,7 +742,7 @@ PHP_METHOD(MongoGridFS, storeFile)
 		char *buf;
 		zval *chunk_id = NULL;
 
-		int chunk_size = size-pos >= global_chunk_size || fp == 0 ? global_chunk_size : size-pos;
+		long chunk_size = size-pos >= global_chunk_size || fp == 0 ? global_chunk_size : size-pos;
```

Comment:
No, but it needs to be done throughout the file. The user also said it didn't quite work for him, so that's why I didn't want to merge it yet either.